pkg/resource/open_id_connect_provider/hooks.go

```diff
@@ -0,0 +1,297 @@
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := desired.ko.DeepCopy()
+
+	rm.setStatusDefaults(ko)
+	if delta.DifferentAt("Spec.Tags") {
+		if err := rm.syncTags(ctx, &resource{ko}); err != nil {
+			return nil, err
+		}
+	}
+	// There really isn't a status of a role... it either exists or doesn't. If
+	// we get here, that means the update was successful and the desired state
+	// of the role matches what we provided...
+	ackcondition.SetSynced(&resource{ko}, corev1.ConditionTrue, nil, nil)
+
+	return &resource{ko}, nil
+}
+
+// compareTags is a custom comparison function for comparing lists of Tag
+// structs where the order of the structs in the list is not important.
+func compareTags(
+	delta *ackcompare.Delta,
+	a *resource,
+	b *resource,
+) {
+	if len(a.ko.Spec.Tags) != len(b.ko.Spec.Tags) {
+		delta.Add("Spec.Tags", a.ko.Spec.Tags, b.ko.Spec.Tags)
+	} else if len(a.ko.Spec.Tags) > 0 {
+		if !commonutil.EqualTags(a.ko.Spec.Tags, b.ko.Spec.Tags) {
+			delta.Add("Spec.Tags", a.ko.Spec.Tags, b.ko.Spec.Tags)
+		}
+	}
+}
+
+// syncTags examines the Tags in the supplied OpenIDConnectProvider and calls the ListOpenIDConnectProviderTags,
+// TagOpenIDConnectProvider and UntagOpenIDConnectProvider API endpoints to ensure that the set of associated Tags stays
+// in sync with the OpenIDConnectProvider.Spec.Tags
+func (rm *resourceManager) syncTags(
+	ctx context.Context,
+	r *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncTags")
+	defer exit(err)
+	toAdd := []*svcapitypes.Tag{}
+	toDelete := []*svcapitypes.Tag{}
+
+	existingTags, err := rm.getTags(ctx, r)
+	if err != nil {
+		return err
+	}
+
+	for _, t := range r.ko.Spec.Tags {
+		if !inTags(*t.Key, *t.Value, existingTags) {
+			toAdd = append(toAdd, t)
+		}
+	}
+
+	for _, t := range existingTags {
+		if !inTags(*t.Key, *t.Value, r.ko.Spec.Tags) {
+			toDelete = append(toDelete, t)
+		}
+	}
+
+	if len(toAdd) > 0 {
+		for _, t := range toAdd {
+			rlog.Debug("adding tag to OpenIDConnectProvider", "key", *t.Key, "value", *t.Value)
+		}
+		if err = rm.addTags(ctx, r, toAdd); err != nil {
+			return err
+		}
+	}
+	if len(toDelete) > 0 {
+		for _, t := range toDelete {
+			rlog.Debug("removing tag from OpenIDConnectProvider", "key", *t.Key, "value", *t.Value)
+		}
+		if err = rm.removeTags(ctx, r, toDelete); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// inTags returns true if the supplied key and value can be found in the
+// supplied list of Tag structs.
+//
+// TODO(jaypipes): When we finally standardize Tag handling in ACK, move this
+// to the ACK common runtime/ or pkg/ repos
+func inTags(
+	key string,
+	value string,
+	tags []*svcapitypes.Tag,
+) bool {
+	for _, t := range tags {
+		if *t.Key == key && *t.Value == value {
+			return true
+		}
```

Comment:
This might panic if a user provides a Key but a not a value. Let's perform a nil check on `t.Value`

Reply:
Done